components/api/src/main/java/com/hotels/styx/api/configuration/ObjectStore.java

```diff
@@ -26,10 +28,17 @@
 public interface ObjectStore<T> {
 
     /**
-     * A repository of named objects.
+     * Returns a named object.
      *
      * @param key object's name
      * @return an object if known
      */
     Optional<T> get(String key);
+
+    /**
+     * Returns a list of full contents.
+     *
+     * @return a collection of all entries.
+     */
+    Collection<Map.Entry<String, T>> entrySet();
```

Comment:
Should we consider using `com.hotels.styx.common.Pair` instead of `Map.Entry`?

Reply:
Possibly. Can you think of any pros/cons using `Pair` vs `Map.Entry`?

Reply:
`Map.Entry` does have a method `setValue(V value)` for mutating it (or throwing `UnsupportedOperationException`), which could give the illusion of mutability, however most Java programmers will be aware of the awkward way that the JCF handles immutable classes.

Otherwise, it doesn't really make a difference, so if `Map.Entry` is more convenient otherwise, I think we can put up with it.